### PR TITLE
[AzureMonitorExporter] Refactor ActivityTagsProcessor for better readability and add benchmarks

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
```
BenchmarkDotNet=v0.13.4, OS=Windows 11 (10.0.22621.1778)
Intel Core i7-8650U CPU 1.90GHz (Kaby Lake R), 1 CPU, 8 logical and 4 physical cores
.NET SDK=7.0.105
  [Host] : .NET 7.0.5 (7.0.523.17405), X64 RyuJIT AVX2
Job=InProcess  Toolchain=InProcessEmitToolchain
|                              Method    |     Mean |   Error |   StdDev |   Median |   Gen0 | Allocated |
|--------------------------------------- |---------:|--------:|---------:|---------:|-------:|----------:|
|     Benchmark_ActivityTagsProcessor    | 282.5 ns | 5.61 ns | 13.66 ns | 279.0 ns | 0.1335 |     560 B |
|  Benchmark_OldActivityTagsProcessor    | 288.3 ns | 5.78 ns | 12.06 ns | 283.5 ns | 0.1335 |     560 B |
| Benchmark_NewDictActivityTagsProcessor | 328.2 ns | 2.82 ns |  2.36 ns | 328.1 ns | 0.1335 |     560 B |
```

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ActivityTagsProcessor.cs [hunk 1]
```diff
@@ -77,75 +83,51 @@ public ActivityTagsProcessor()
...
+                        AzMonList.Add(ref UnMappedTags, new KeyValuePair<string, object?>(tag.Key, array.ToCommaDelimitedString()));
+                        continue;
+                    }
+
                     AzMonList.Add(ref UnMappedTags, tag);
                 }
+            }
 
-                previousActivityType = _currentActivityType;
+            if ((activityType & OperationType.Azure) == OperationType.Azure)
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ActivityTagsProcessor.cs:127]:
```suggestion
            if ((activityType.HasFlag(OperationType.Azure)))
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TagsTests.cs [hunk 2]
```diff
@@ -108,13 +109,11 @@ public void TagObjects_Mapped()
             activityTagsProcessor.CategorizeTags(activity);
 
             Assert.Equal(OperationType.Http, activityTagsProcessor.activityType);
-            Assert.Equal(4, activityTagsProcessor.MappedTags.Length);
+            Assert.Equal(6, activityTagsProcessor.MappedTags.Length);
             Assert.Equal("https", AzMonList.GetTagValue(ref activityTagsProcessor.MappedTags, SemanticConventions.AttributeHttpScheme));
             Assert.Equal("localhost", AzMonList.GetTagValue(ref activityTagsProcessor.MappedTags, SemanticConventions.AttributeHttpHost));
             Assert.Equal("8888", AzMonList.GetTagValue(ref activityTagsProcessor.MappedTags, SemanticConventions.AttributeHttpHostPort));
             Assert.Equal("127.0.0.1", AzMonList.GetTagValue(ref activityTagsProcessor.MappedTags, SemanticConventions.AttributeNetHostIp));
-            Assert.Single(activityTagsProcessor.UnMappedTags);
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TagsTests.cs:116]: was this a bug?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ActivityTagsProcessor.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ActivityTagsProcessor.cs:127]: I intentionally used this syntax, as I was removing `OperationType.Azure`, using this one - `activityType ^= OperationType.Azure;`.  If you think using HasFlag would improve readability, please let me know

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ActivityTagsProcessor.cs:127]: Not a blocker for me.
I think `HasFlag` is easier to read, but it's doing this same operation. :)
https://github.com/dotnet/runtime/blob/53af66a48215f0b7833dce499343eccae1dc9dff/src/libraries/System.Private.CoreLib/src/System/Enum.cs#L400

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ActivityTagsProcessor.cs:127]: After you point me to the code, I'm thinking why to use `HasFlag`. Current code will work better as it reduces few lines of code executing within `HasFlag` method. :blush:

**Comment by rajkumar-rangaraj** [comment]
I could resolve conflict, results are better now. (Removed one HttpAttribute and added enduserid to benchmarks).

```
|                          Method |     Mean |   Error |  StdDev |   Gen0 | Allocated |
|-------------------------------- |---------:|--------:|--------:|-------:|----------:|
| Benchmark_ActivityTagsProcessor | 267.3 ns | 4.28 ns | 3.57 ns | 0.1335 |     560 B
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ActivityTagsProcessor.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ActivityTagsProcessor.cs]: Removed this code to make it easier.

**Review by vishweshbankwar — APPROVED** [review]

**Pull request merged**